nixos/modules/services/misc/beanstalkd.nix

```diff
@@ -0,0 +1,51 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.beanstalkd;
+  pkg = pkgs.beanstalkd;
+in
+
+{
+  # interface
+
+  options = {
+    services.beanstalkd = {
+      enable = mkEnableOption "Enable the Beanstalk work queue.";
+
+      listen = {
+        port = mkOption {
+          type = types.int;
+          description = "TCP port that will be used to accept client connections.";
+          default = 11300;
+        };
+
+        address = mkOption {
+          type = types.str;
+          description = "IP address to listen on.";
+          default = "0.0.0.0";
+        };
+      };
+    };
+  };
+
+  # implementation
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ pkg ];
+
+    systemd.services.beanstalkd = {
+      description = "Beanstalk Work Queue";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        DynamicUser = true;
+        Restart = "always";
+        ExecStart = "${pkg}/bin/beanstalkd -l ${cfg.listen.address} -p ${toString cfg.listen.port}";
+      };
```

Comment:
@flokli are there any more options that could be used to sandbox the service even more?

Reply:
From what I read, beanstalkd doesn't really use local files, so you could use
[TemporaryFileSystem=](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#TemporaryFileSystem=) to mask out most of the filesystem too.

Apart from that, you could look into [SystemCallFilter=](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#SystemCallFilter=), and some [resource limits](https://www.freedesktop.org/software/systemd/man/systemd.resource-control.html).

Reply:
That sounds great! Unfortunately I'm not overly familiar with beanstalkd and the inner workings (I have an old application currently running on Debian which uses beanstalkd and I'm trying to move the app to NixOS), or setting up limitations with systemd to this extent. Do you have specific suggestions as to what limitations I should add?